server/AuthController.js

```diff
@@ -12,6 +13,42 @@ module.exports = {
     res.set(JWT.renewTokenHeader, token);
     res.status(200).send({ token })
   },
+  impersonate: (_grabUser, id='id') => async (req, res) => {
+    try {
+      let token = JWT.issue({
+        // The new user will be whatever the target id is
+        user: req.param(id),
+        // Add the previous user to the back of the stack of users ids
+        // Makes it easy to use with lodash :D
+        impersonateStack: _.concat(req.user.id)(req.tokenPayload.impersonateStack || [])
```

Comment:
Use spread `impersonateStack: [req.user.id, ...req.tokenPayload.impersonateStack]`

You _might_ need the `|| []` after the impersonate stack